xla/service/gpu/tests/gemm_rewrite_test.cc

```diff
@@ -4651,36 +4695,42 @@ TEST_P(ParameterizedFp8GemmRewriteTest, UnsupportedTypesF8) {
 ; CHECK-DAG:         }
 ; CHECK-DAG:         "epilogue":"DEFAULT"
 ; CHECK:           }
-; CHECK:         ROOT [[OUT:%[^ ]+]] = f8e5m2[16,16]{1,0} convert
-      )",
-                                                replacements_));
+; CHECK:         ROOT [[OUT:%[^ ]+]] = <<F8E5M2_DATATYPE_PLACEHOLDER>>[16,16]{1,0} convert
+      )", replacements_));
 }
 
 TEST_P(ParameterizedFp8GemmRewriteTest, UnscaledABUnscaledDF8) {
 #if GOOGLE_CUDA && CUDA_VERSION < 12000
   GTEST_SKIP() << "F8 gemm rewrite is only supported in CUDA 12 and above.";
 #endif  // CUDA_VERSION < 12000
-  const char* hlo_text = R"(
+
+#if TENSORFLOW_USE_ROCM && TF_ROCM_VERSION < 60000
+  GTEST_SKIP() << "F8 gemm rewrite is only supported in ROCm 6.0 and above.";
+#endif // TF_ROCM_VERSION < 60000
+
+  const char* raw_hlo_text = R"(
     HloModule test
 
     ENTRY test {
-      x = f8e4m3fn[16,32] parameter(0)
-      y = f8e4m3fn[32,16] parameter(1)
-      ROOT out = f8e4m3fn[16,16] dot(x, y), lhs_contracting_dims={1}, rhs_contracting_dims={0}
+      x = <<F8E4M3_DATATYPE_PLACEHOLDER>>[16,32] parameter(0)
+      y = <<F8E4M3_DATATYPE_PLACEHOLDER>>[32,16] parameter(1)
+      ROOT out = <<F8E4M3_DATATYPE_PLACEHOLDER>>[16,16] dot(x, y), lhs_contracting_dims={1}, rhs_contracting_dims={0}
           }
 
 )";
 
+  std::string hlo_text = absl::StrReplaceAll(raw_hlo_text, replacements_);
   CheckFp8IfSupported(hlo_text);
   // auto comp = se::DeviceCom
-  RunAndFilecheckHloRewrite(hlo_text, GemmRewriter(CudaHopperOrRocm()),
-                            R"(
-; CHECK-LABEL: ENTRY %test ({{.*}}: f8e4m3fn[16,32], {{.*}}: f8e4m3fn[32,16]) -> f8e4m3fn[16,16] {
-; CHECK-NEXT:    [[P0:%[^ ]+]] = f8e4m3fn[16,32]{1,0} parameter(0)
-; CHECK-NEXT:    [[P1:%[^ ]+]] = f8e4m3fn[32,16]{1,0} parameter(1)
-; CHECK-NEXT:    [[P1_TRANSPOSE:%[^ ]+]] = f8e4m3fn[16,32]{1,0} transpose([[P1]]), dimensions={1,0}
+  RunAndFilecheckHloRewrite(hlo_text, GemmRewriter(CudaHopperOrRocmMI300()),
+                            absl::StrReplaceAll(R"(
+; CHECK-LABEL: ENTRY %test ({{.*}}: <<F8E4M3_DATATYPE_PLACEHOLDER>>[16,32], {{.*}}: <<F8E4M3_DATATYPE_PLACEHOLDER>>[32,16]) -> <<F8E4M3_DATATYPE_PLACEHOLDER>>[16,16] {
+; CHECK-NEXT:    [[P0:%[^ ]+]] = <<F8E4M3_DATATYPE_PLACEHOLDER>>[16,32]{1,0} parameter(0)
+; CHECK-NEXT:    [[P1:%[^ ]+]] = <<F8E4M3_DATATYPE_PLACEHOLDER>>[32,16]{1,0} parameter(1)
+; CHECK-NEXT:    [[P1_TRANSPOSE:%[^ ]+]] = <<F8E4M3_DATATYPE_PLACEHOLDER>>[16,32]{1,0} transpose([[P1]]), dimensions={1,0}
 ; CHECK-NEXT:    [[C1:[^ ]+]] = f32[] constant(1)
-; CHECK-NEXT:    ROOT [[OUT:%[^ ]+]] = f8e4m3fn[16,16]{1,0} custom-call([[P0]], [[P1_TRANSPOSE]], [[C1]], [[C1]], [[C1]], /*index=5*/[[C1]]),
+; CHECK-GCN-NEXT:    [[OUT:%[^ ]+]] = f32[16,16]{1,0} custom-call([[P0]], [[P1_TRANSPOSE]], [[C1]], [[C1]], [[C1]], /*index=5*/[[C1]]),
+; CHECK-PTX-NEXT:    ROOT [[OUT:%[^ ]+]] = <<F8E4M3_DATATYPE_PLACEHOLDER>>[16,16]{1,0} custom-call([[P0]], [[P1_TRANSPOSE]], [[C1]], [[C1]], [[C1]], /*index=5*/[[C1]]),
```

Comment:
@cheshire, @ezhulenev, any opinions on using CHECK-GCN-NEXT  and CHECK-PTX-NEXT like this? I'm unsure if this is a good idea or not. `filecheck.cc` is modified to have `CHECK-GCN-NEXT` on match for rocm and `CHECK-PTX-NEXT` only match for CUDA, to account for slightly difference in how gemm_rewriter operates between the two.